File: 프로그래머스/unrated/136798. 기사단원의 무기/기사단원의 무기.java
class Solution {
    public int solution(int number, int limit, int power) {
        if(number==1) return 1;
        int answer = 1;
        int cnt = 0;
        int divisor = 0;
        for(int i=2; i<=number; i++){
            cnt = 0;
            for(int j=1; j*j<=i; j++){
                if(i%j==0) cnt+=2;
                if(j*j==i) cnt--;
                if(cnt>limit){
                    cnt = power;
                    break;
                }
            }
            answer += cnt;
        }
        return answer;
    }
}

File: 프로그래머스/unrated/136798. 기사단원의 무기/기사단원의 무기.js
function solution(number, limit, power) {
    var answer = 0;
    for(var i=1; i<=number; i++){
        var measure = 0;
        for(var j=1; j<=Math.sqrt(i); j++){
            if(i%j==0) {
                if(i/j === j) measure++;
                else measure+=2;
            }
        }
        answer += measure > limit ? power : measure;
    }
    return answer;
}